File: app/src/main/java/ru/profapp/RanobeReader/JsonApi/JsonBaseClass.java
package ru.profapp.RanobeReader.JsonApi;

import org.jsoup.nodes.Document;

import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.ExecutionException;

import ru.profapp.RanobeReader.Common.ErrorConnectionException;
import ru.profapp.RanobeReader.Helpers.HtmlParser;
import ru.profapp.RanobeReader.Helpers.StringHelper;
import ru.profapp.RanobeReader.Helpers.UrlParser;

class JsonBaseClass {

    Map<String, String> Cookies = new HashMap<>();

    String getUrlText(String request) throws ErrorConnectionException {

        try {
            String result = new UrlParser().execute(request).get();
            if (result == null) {
                throw new ErrorConnectionException();
            }
            return result;

        } catch (InterruptedException | ExecutionException ignored) {

        } catch (NullPointerException e) {
            throw new ErrorConnectionException(e);
        }
        return "";
    }

    String getUrlText(String request,Map<String, String> header) throws ErrorConnectionException {

        try {
            String result = new UrlParser(header).execute(request).get();
            if (result == null) {
                throw new ErrorConnectionException();
            }
            return result;

        } catch (InterruptedException | ExecutionException ignored) {

        } catch (NullPointerException e) {
            throw new ErrorConnectionException(e);
        }
        return "";
    }

    String getDocumentText(Map<String, String> Cookies, String... params)
            throws ErrorConnectionException {

        Document html;
        try {
            html = new HtmlParser(Cookies).execute(params).get();
            String result = html.body().html();
            return StringHelper.getInstance().cleanJson(result);
        } catch (InterruptedException | ExecutionException ignored) {
        } catch (NullPointerException e) {
            throw new ErrorConnectionException(e);
        }
        return "";
    }

    String getDocumentText(Map<String, String> Cookies, Map<String, String> Data,
            Map<String, String> Header,
            String... params) throws ErrorConnectionException {

        Document html;
        try {
            html = new HtmlParser(Cookies, Data, Header).execute(params).get();

            String result = html.body().html();
            return StringHelper.getInstance().cleanJson(result);
        } catch (InterruptedException | ExecutionException ignored) {

        } catch (NullPointerException e) {
            throw new ErrorConnectionException(e);
        }
        return "";
    }
}

File: app/src/main/java/ru/profapp/RanobeReader/JsonApi/JsonBaseClass.kt
package ru.profapp.RanobeReader.JsonApi

import org.jsoup.nodes.Document
import ru.profapp.RanobeReader.Common.ErrorConnectionException
import ru.profapp.RanobeReader.Helpers.HtmlParser
import ru.profapp.RanobeReader.Helpers.StringHelper
import ru.profapp.RanobeReader.Helpers.UrlParser
import java.util.*
import java.util.concurrent.ExecutionException

open class JsonBaseClass {

    var Cookies: Map<String, String> = HashMap()


    fun getUrlText(request: String): String {

        try {
            return UrlParser().execute(request).get() ?: throw ErrorConnectionException()

        } catch (ignored: InterruptedException) {

        } catch (ignored: ExecutionException) {
        } catch (e: NullPointerException) {
            throw ErrorConnectionException(e)
        }

        return ""
    }


    fun getUrlText(request: String, header: Map<String, String>): String {

        try {
            return UrlParser(header).execute(request).get() ?: throw ErrorConnectionException()

        } catch (ignored: InterruptedException) {
        } catch (ignored: ExecutionException) {
        } catch (e: NullPointerException) {
            throw ErrorConnectionException(e)
        }

        return ""
    }


    fun getDocumentText(Cookies: Map<String, String>, vararg params: String): String {

        val html: Document
        try {
            html = HtmlParser(Cookies).execute(*params).get()
            val result = html.body().html()
            return StringHelper.cleanJson(result)
        } catch (ignored: InterruptedException) {
        } catch (ignored: ExecutionException) {
        } catch (e: NullPointerException) {
            throw ErrorConnectionException(e)
        }

        return ""
    }


    fun getDocumentText(Cookies: Map<String, String>, Data: Map<String, String>,
                        Header: Map<String, String>,
                        vararg params: String): String {

        val html: Document
        try {
            html = HtmlParser(Cookies, Data, Header).execute(*params).get()

            val result = html.body().html()
            return StringHelper.cleanJson(result)
        } catch (ignored: InterruptedException) {

        } catch (ignored: ExecutionException) {
        } catch (e: NullPointerException) {
            throw ErrorConnectionException(e)
        }

        return ""
    }
}
